src/main/java/com/box/sdk/BoxSearchParameters.java

```diff
@@ -0,0 +1,427 @@
+        return this.ancestorFolderIds;
+    }
+    /**
+     * Set the list of folder id's to be applied as a filter on the search filters.
+     * @param ancestorFolderIds a list of folder ids that will contain results to specific folders.
+     */
+    public void setAncestorFolderIds(List<String> ancestorFolderIds) {
+        this.ancestorFolderIds = ancestorFolderIds;
+    }
+    /**
+     * Return content types that or being applied as a filter (name,description,tags,file_content).
+     * @return  contentTypes.
+     */
+    public List<String> getContentTypes() {
+        return this.contentTypes;
+    }
+    /**
+     * Set list of content types that will be used as a filters.
+     * @param contentTypes a list of content types such as (name,description,tags,file_content).
+     */
+    public void setContentTypes(List<String> contentTypes) {
+        this.contentTypes = contentTypes;
+    }
+    /**
+     * Return the type you want to return in your search.
+     * @return String type.
+     */
+    public String getType() {
+        return this.type;
+    }
+    /**
+     * Set the type you want to search for can be file, folder, or web_link.
+     * @param type can be file, folder, or web_link.
+     */
+    public void setType(String type) {
+        this.type = type;
+    }
+    /**
+     * Return the specified trash search preference.
+     * @return String trashContent.
+     */
+    public String getTrashContent() {
+        return this.trashContent;
+    }
+    /**
+     * Set trash filter. Can be trashed_only or non_trashed_only, without this parameter default to non_trashed_only.
+     * @param trashContent Can be trashed_only or non_trashed_only.
+     */
+    public void setTrashContent(String trashContent) {
+        this.trashContent = trashContent;
+    }
+    /**
+     * Retrieve the existing BoxMetaDataFilterRequest.
+     * @return this.BoxMetaDataFilterRequest
+     */
+    public BoxMetadataFilterRequest getMdFilters() {
+        BoxMetadataFilterRequest bmfr = new BoxMetadataFilterRequest();
+        for (BoxMetadataFilter bmf : this.mdFilters) {
+            bmfr.addBoxMetadataFilter(bmf);
+        }
+        return bmfr;
+    }
+    /**
+     * Set the current list of Metadata Filters.
+     * @param mdFilters a list of the current metadata filters.
+     */
+    public void setMdFilters(List<BoxMetadataFilter> mdFilters) {
+        this.mdFilters = mdFilters;
+    }
+    /**
+     * Formated String is returned.
+     * @param    Object that will be printed.
+     * @return this.String
+     */
+    private String printParam(Object obj)    {
+        return obj.toString().replaceAll("[\\[\\]]", "");
+    }
+    /**
+     * Checks String to see if the parameter is null.
+     * @param    paramValue Object that will be checked if null.
+     * @return this.true if the parameter that is being checked is not null
+     */
+    public boolean isNotNull(Object paramValue) {
```

Comment:
This method is kind of confusing because it does more than just check for null. I'd recommend just using `==` when checking for null objects and changing this method to `isNullOrEmpty(String)` if you need a helper method for strings.